internal_transfers/actions/src/parse.ts

```diff
@@ -0,0 +1,55 @@
+import { SettlementEvent, TradeEvent, TransferEvent } from "./models";
+import { Log } from "@tenderly/actions";
+import { addressFromBytes32, transferInvolves } from "./utils";
+
+export const SETTLEMENT_CONTRACT_ADDRESS =
+  "0x9008d19f58aabd9ed0d60971565aa8510560ab41";
+export const SETTLEMENT_EVENT_TOPIC =
+  "0x40338ce1a7c49204f0099533b1e9a7ee0a3d261f84974ab7af36105b8c4e9db4";
```

Comment:
I'm not a fan of hard coding those event topics and manually parsing the topics and data. I know this is how it was done in the alerter as well, but it's not really clean for a larger project (at the end of a day a subtle bug in the slice parsing logic caused the alerter to not fire yesterday). 

The better way to do this would - I believe - be to depend on https://www.npmjs.com/package/@cowprotocol/contracts and use the contract ABI and ethers.js parsing library to get typed objects with the relevant fields (see [this tutorial](https://medium.com/@kaishinaw/ethereum-logs-hands-on-with-ethers-js-a28dde44cbb6)).

This way you don't need to rewrite the parsing code.

Given it was not done in the other project, I don't think this is blocking, but I think it would be nice to have an example of how to properly parse these events using libraries rather than custom code.

Reply:
This seems like something we can and should do, but also I could create an issue for it and circle back as an improvement. My python implementation did what you are suggesting, not sure why I went this route here.

Reply:
One foreseeable issue I have with this suggested approach is for the vendoring of transfer artifacts. Sure it might be lightweight to get the Trade and Settlement Event data from our contract suite, but for erc20 token ABI we would have to import all of openzeppelin! Is there some less annoying way to just get that one thing? Only idea I can think of is to write my own `iERC20.sol` then depend on some solidity compiler and have a whole other build step just to generate the ABI... Even this seems ridiculous, why not just put the abi in the project directly?

Reply:
The contracts package should have the ABI for OpenZeppelin's ERC20 tokens already.

Alternatively, you can add what you need to vendor to the repo, find a reputable source for it online, and offer an easy way to verify that the vendored version matches the real one. [This](https://github.com/cowprotocol/token/tree/main/src/contracts/vendored) is what we did to vendor a few contracts from OpenZeppelin in the token repo.

Reply:
Here we have a PR based on this which parses Events using contract artifacts and ethers.

https://github.com/cowprotocol/solver-rewards/pull/193

I can merge that into this and then this into its original destination or whatever really.

Reply:
This has been resolved by #193.